emr-user-role-mapper-application/src/main/java/com/amazon/aws/emr/credentials/STSCredentialsProvider.java

```diff
@@ -49,10 +52,27 @@
         });
 
     synchronized static AWSSecurityTokenService getStsClient() {
+        try {
+            if (region == null) {
+                region = Regions.getCurrentRegion();
+            }
+        } catch (Exception e){
+            log.info("Could not determine the AWS region!");
+        }
```

Comment:
Move this inside the if check for stsClient.

Reply:
This has been addressed.